### PR TITLE
Remove suffix `.aux` from progName when installed by cs

### LINKED ISSUE #1662: Confusing path shown in --usage output
**Version(s)**

0.1.18 installed via `cs install scala-experimental`

**Describe the bug**

When `scala` shows the “usage” instructions it shows a confusing path for the `scala` executable itself:

~~~ text
$ scala test --usage
Usage: /home/julien/.local/share/coursier/bin/.scala.aux test [options]
~~~

**To Reproduce**

- Install `scala` via `cs install scala-experimental`
- Run `scala test --usage`

**Expected behaviour**

~~~ text
$ scala test --usage
Usage: scala test [options]
~~~

### PR DESCRIPTION
**Pull request by lwronski**
Fixes #1662.

### CODE REVIEW

File: modules/cli/src/main/scala/scala/cli/ScalaCli.scala [hunk 1]
```diff
@@ -21,7 +21,15 @@ object ScalaCli {
     // the Scala CLI native image, no need to manually load it.
     coursier.jniutils.LoadWindowsLibrary.assumeInitialized()
 
-  val progName = (new Argv0).get("scala-cli")
+  val progName = {
+    val argv0 = (new Argv0).get("scala-cli")
+    val idx   = argv0.lastIndexOf(File.separator)
+    val last  = if (idx < 0) argv0 else argv0.drop(idx + 1)
```
Comment (tgodzik) [inline modules/cli/src/main/scala/scala/cli/ScalaCli.scala]:
```suggestion
    val last  = Paths.get(argv0).getFileName().toString()
```
?

File: modules/integration/src/test/scala/scala/cli/integration/SipScalaTests.scala [hunk 2]
```diff
@@ -107,6 +107,19 @@ class SipScalaTests extends ScalaCliSuite {
...
+  if (TestUtil.isNativeCli)
+    test(s"usage instruction should point to scala when installing by cs") { // https://github.com/VirtusLab/scala-cli/issues/1662
+      TestInputs.empty.fromRoot {
+        root => // cs install binaries under .app-name.aux and scala-cli should drop .aux from progName
+          val binary       = "scala".prepareBinary(root)
+          val csBinaryName = root / ".scala.aux"
+          os.move(binary, csBinaryName)
+          val output   = os.proc(csBinaryName, "test", "--usage").call(check = false).out.text()
+          val usageMsg = TestUtil.removeAnsiColors(output)
+          expect(usageMsg.contains(" scala "))
```
Comment (julienrf) [inline modules/integration/src/test/scala/scala/cli/integration/SipScalaTests.scala]: Maybe this assertion is too weak? Should it be something like `expect(!(usageMsg.contains(".scala.aux")))`?

Comment (lwronski) [reply modules/integration/src/test/scala/scala/cli/integration/SipScalaTests.scala]: Right, Thanks for review!

File: modules/integration/src/test/scala/scala/cli/integration/SipScalaTests.scala [hunk 3]
```diff
@@ -107,6 +107,20 @@ class SipScalaTests extends ScalaCliSuite {
...
+          expect(!usageMsg.contains(".scala.aux"))
```
Comment (julienrf) [inline modules/integration/src/test/scala/scala/cli/integration/SipScalaTests.scala]: What is the entire message? I think it should be exactly:

> Usage: scala test [options]

Is this possible?

Comment (lwronski) [reply modules/integration/src/test/scala/scala/cli/integration/SipScalaTests.scala]: If `case-app` doesn't change the message for usage options, it should be exactly `Usage: scala test [options]`.

But for now, it should be exactly `Usage: scala test [options]`, so I modify integration test.

**Review by julienrf — APPROVED** [review]
Thank you Łukas!

File: modules/cli/src/main/scala/scala/cli/ScalaCli.scala [hunk 4]
```diff
@@ -21,7 +22,14 @@ object ScalaCli {
     // the Scala CLI native image, no need to manually load it.
     coursier.jniutils.LoadWindowsLibrary.assumeInitialized()
 
-  val progName = (new Argv0).get("scala-cli")
+  val progName = {
+    val argv0 = (new Argv0).get("scala-cli")
+    val last  = Paths.get(argv0).getFileName.toString
+    last match {
+      case s".${name}.aux" => name // cs install binaries under .app-name.aux and add to the PATH
```
Comment (Gedochao) [inline modules/cli/src/main/scala/scala/cli/ScalaCli.scala]: NIT
```suggestion
      case s".${name}.aux" => name // cs installs binaries under .app-name.aux and adds them to the PATH
```

**Review by Gedochao — APPROVED** [review]
LGTM

**Pull request merged**